dali/pipeline/data/tensor_list.h

```diff
@@ -646,24 +646,24 @@ class DLL_PUBLIC TensorList {
     return {data_.capacity()};
   }
 
-  /**
-   * @brief Set the type of the TensorList. The type needs to be set before calling
-   * the Resize function that gives the shape. Type can be changed, if the current storage
-   * is not big enough, the memory will be reallocated.
+  /** @defgroup SetTypeFunctions Type setting functions.
+   *
+   * Set the type of the current batch. It must be set before calling Resize(shape).
+   * It cannot be used to change the type after allocation happened.
+   *
+   * Resize(shape, new_type) can be used without prior set_type call or to request a different type
+   * after allocation.
+   * @{
    */
   inline void set_type(const DALIDataType new_type_id) {
     data_.set_type(new_type_id);
   }
 
-  /**
-   * @brief Set the type of the TensorList. The type needs to be set before calling
-   * the Resize function that gives the shape. Type can be changed, if the current storage
-   * is not big enough, the memory will be reallocated.
-   */
   template <typename T>
   inline void set_type() {
     data_.set_type(TypeTable::GetTypeId<T>());
   }
+  /** @} */  // end of SetTypeFunctions
```

Comment:
```suggestion
  ///@}
```
Block comments for single line are a retreat to the '70s...

Reply:
And consistent use of doxygen in our codebase.